port_ocean/helpers/async_client.py

```diff
@@ -32,25 +34,34 @@ def _init_transport(  # type: ignore[override]
         if transport is not None:
             return super()._init_transport(transport=transport, **kwargs)
 
-        return self._transport_class(
-            wrapped_transport=httpx.AsyncHTTPTransport(
-                **kwargs,
-            ),
-            logger=logger,
-            **(self._transport_kwargs or {}),
-        )
+        if self._retry_config is not None:
+            return self._transport_class(
+                wrapped_transport=httpx.AsyncHTTPTransport(**kwargs),
+                retry_config=self._retry_config,
+                logger=logger,
+            )
+        else:
+            return self._transport_class(
+                wrapped_transport=httpx.AsyncHTTPTransport(**kwargs),
+                logger=logger,
+                **(self._transport_kwargs or {}),
+            )
```

Comment:
```suggestion
            return self._transport_class(
                wrapped_transport=httpx.AsyncHTTPTransport(**kwargs),
                retry_config=self._retry_config,
                logger=logger,
            )
```

I don't think there's a reason to add this "if" here